[quant][gx][graphmode][refactor] Remove modules from Quantizer

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59042 [quant][graphmode][fx][refactor] Remove _convert from Quantizer class
* #59041 [quant][graphmode][fx][fix] Fix support for custom module
* #59040 [quant][graphmode][fx][refactor] Remove Quantizer class from convert (QuantizeHandler)
* #59039 [quant][graphmode][fx][refactor] Remove Quantizer class from quantize_node
* #59038 [quant][graphmode][fx][refactor] Remove _prepare from Quantizer class
* #59037 [quant][graphmode][fx][refactor] Remove find_matches from Quantizer class
* #59036 [quant][graphmode][fx][refactor] Remove fold_weights from Quantizer class
* #59035 [quant][graphmode][fx][refactor] Remove save_state and restore_state from Quantizer class
* #59034 [quant][graphmode][fx][refactor] Remove prepare_custom_config from Quantizer class
* #59033 [quant][graphmode][fx][refactor] Remove patterns from Quantizer class
* #59032 [quant][graphmode][fx][refactor] Remove node_name_to_scope from Quantizer
* #59031 [quant][graphmode][fx][refactor] Remove quantized_graph from Quantizer
* **#59030 [quant][gx][graphmode][refactor] Remove modules from Quantizer**
* #59029 [quant][graphmode][fx] Factor out run_weight_observer
* #59028 [quant][graphmode][fx] Merge quant_env and env

Summary:
Trying to remove Quantizer class and split prepare and convert code

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724875](https://our.internmc.facebook.com/intern/diff/D28724875)

jerryzh168 committed:
[quant][gx][graphmode][refactor] Remove modules from Quantizer

Summary:
Trying to remove Quantizer class and split prepare and convert code

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724875).

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
in a future PR, could we consider removing this altogether and doing the lookup dynamically?

jerryzh168 committed:
Update on "[quant][gx][graphmode][refactor] Remove modules from Quantizer"


Summary:
Trying to remove Quantizer class and split prepare and convert code

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724875](https://our.internmc.facebook.com/intern/diff/D28724875)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724875).

jerryzh168 committed:
Update on "[quant][gx][graphmode][refactor] Remove modules from Quantizer"


Summary:
Trying to remove Quantizer class and split prepare and convert code

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724875](https://our.internmc.facebook.com/intern/diff/D28724875)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724875).

jerryzh168 committed:
Update on "[quant][gx][graphmode][refactor] Remove modules from Quantizer"


Summary:
Trying to remove Quantizer class and split prepare and convert code

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724875](https://our.internmc.facebook.com/intern/diff/D28724875)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724875).
